Source/SwiftLintBuiltInRules/Rules/Idiomatic/RedundantTypeAnnotationRule.swift

```diff
@@ -1,9 +1,143 @@
+        )
+    }
+}
+
+private extension RedundantTypeAnnotationRule {
+    final class Visitor: ViolationsSyntaxVisitor<ConfigurationType> {
+        override func visitPost(_ node: VariableDeclSyntax) {
+            guard node.isViolation(for: configuration),
+                  let binding = node.bindings.last,
+                  let typeAnnotation = binding.typeAnnotation
+            else {
+                return
+            }
+
+            violations.append(typeAnnotation.positionAfterSkippingLeadingTrivia)
+        }
+    }
+
+    final class Rewriter: ViolationsSyntaxRewriter {
+        private let configuration: RedundantTypeAnnotationConfiguration
+
+        init(configuration: RedundantTypeAnnotationConfiguration,
+             locationConverter: SourceLocationConverter,
+             disabledRegions: [SourceRange]) {
+            self.configuration = configuration
+
+            super.init(locationConverter: locationConverter, disabledRegions: disabledRegions)
+        }
+
+        override func visit(_ node: VariableDeclSyntax) -> DeclSyntax {
+            guard node.isViolation(for: configuration),
+                  let lastBinding = node.bindings.last,
+                  let typeAnnotation = lastBinding.typeAnnotation,
+                  let initializer = lastBinding.initializer
+            else {
+                return super.visit(node)
+            }
+
+            correctionPositions.append(typeAnnotation.positionAfterSkippingLeadingTrivia)
+
+            // Add a leading whitespace to the initializer sequence so there is one
+            // between the variable name and the '=' sign
+            let initializerWithLeadingWhitespace = initializer
+                .with(\.leadingTrivia, Trivia.space)
+            // Set the type annotation of the last binding to nil to remove redundancy
+            let lastBindingWithoutTypeAnnotation = lastBinding
+                .with(\.typeAnnotation, nil)
+                .with(\.initializer, initializerWithLeadingWhitespace)
+
+            return super.visit(node.with(
+                \.bindings,
+                node.bindings.dropLast() + [lastBindingWithoutTypeAnnotation]
+            ))
+        }
+    }
+}
+
+private extension VariableDeclSyntax {
+    func isViolation(for configuration: RedundantTypeAnnotationConfiguration) -> Bool {
+        // Checks if none of the attributes flagged as ignored in the configuration
+        // are set for this declaration
+        let doesNotContainIgnoredAttributes = configuration.ignoredAnnotations.allSatisfy {
+            !self.attributes.contains(attributeNamed: $0)
+        }
+
+        // Only take the last binding into account in case multiple
+        // variables are declared on a single line.
+        // This binding must have both a type declaration and an initializer
+        // sequence for it to be potentially redundant.
+        guard doesNotContainIgnoredAttributes,
+              let binding = bindings.last,
```

Comment:
Might be fine for now, but actually restricting the rule to the last binding only is not necessary. It could as well check all other bindings.